.github/workflows/build.yml

```diff
@@ -0,0 +1,26 @@
+name: Build gwt-site-webapp
+on:
+  push:
+    branches: ['main']
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Compile the webapp to JS
+        run: mvn --batch-mode -e verify
```

Comment:
nit: add `--no-transfer-progress` (or `-ntp`) to make the logs less verbose and more readable